package_bag/tests.py

```diff
@@ -25,6 +28,56 @@
 RIGHTS_FIXTURE_DIR = join(settings.BASE_DIR, 'fixtures', 'rights')
 
 
+class TestHelpers(TestCase):
+
+    def test_expected_filename(self):
+        passing_filenames = ["7d24b2da347b48fe9e59d8c5d4424235.tar", "4b4334fba43a4cf4940f6c8e6d892f60.tar.gz"]
+        failing_filenames = ["6163b89b00bb4c5cbfc50eb34cb49a75", "example_bag.tar"]
+        for filename in passing_filenames:
+            self.assertTrue(expected_file_name(filename))
+        for filename in failing_filenames:
+            self.assertFalse(expected_file_name(filename))
+
+
+class TestS3Download(TestCase):
+    fixtures = [join(settings.BASE_DIR, "fixtures", "s3_download.json")]
+
+    @patch("package_bag.routines.S3ObjectDownloader.__init__")
+    @mock_s3
+    def test_get_list_to_download(self, mock_init):
+        """Test list contains one filename that is in the database"""
+        region_name, access_key, secret_key, bucket = settings.S3
+        mock_init.return_value = None
+        object_downloader = S3ObjectDownloader()
+        object_downloader.src_dir = settings.SRC_DIR
+        s3 = boto3.resource(service_name='s3', region_name=region_name, aws_access_key_id=access_key, aws_secret_access_key=secret_key)
+        s3.create_bucket(Bucket=bucket)
+        s3_client = boto3.client('s3', region_name=region_name)
+        object_downloader.bucket = s3.Bucket(bucket)
```

Comment:
These lines seem to be repeated; it is worth pulling them out to a helper? Alternatively, it seems like you're jumping through a lot of hoops because you're mocking the `__init__` method. Is there a way to NOT mock init?

Reply:
I do need to create a mock S3 instance regardless, which is where most of this is coming from. I wasn't sure if I was able to set up the mock values in setUp, or otherwise inherit there from somewhere?

Reply:
OK I simplified this a bit.